### PR TITLE
Split pipelines

### PR DESCRIPTION
**Pull request by JoshLove-msft**
The goal of this PR is to validate the approach needed to split the function extensions and client library pipelines into separate pipelines to help facilitate easier releases/testing/maintenance of these separate packages.

Once this merges and we verify there are no issues I will do a follow up PR for the other relevant pipelines from https://github.com/Azure/azure-sdk-for-net/issues/38271

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/tests.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ extends:
   parameters:
     MaxParallel: 4
     ServiceDirectory: servicebus
-    SDKType: client
+    Project: Azure.Messaging.ServiceBus
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/tests.yml:8]: The CI template doesn't have the Project parameter, so I'm not sure if there is a way to limit the CI to only running the tests for a specific project. 
/cc @benbp

**Review by jsquire — APPROVED** [review]
Yay!

**Review by m-redding — APPROVED** [review]

**Pull request merged**